simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,400 @@
+"""
+Generates data for the simple-acquaintances PSL example.
+
+Use -h/--help for usage options.
+"""
+
+import argparse
+import json
+import numpy as np
+import os
+
+from itertools import combinations
+from scipy.stats import truncnorm
+from sklearn.metrics.pairwise import cosine_similarity
+
+DEFAULT_GLOBAL_LIKES = 5
+DEFAULT_LOCAL_LIKES = 5
+DEFAULT_LOCAL_LIKES_VARIANCE = 1.0
+DEFAULT_MAX_RAND_INT = 1000000
+DEFAULT_PEOPLE = 25
+DEFAULT_PLACES = 5
+DEFAULT_PLACES_LIVED_MEAN = 3.0
+DEFAULT_PLACES_LIVED_SD = 1.0
+DEFAULT_TARGET_SPLIT = 0.2
+
+
+def checkNonNegativeInt(arg):
+    """
+    Check if argument is non negative integer.
+    """
+    if not arg.isdigit():
+        raise argparse.ArgumentTypeError(
+            '{} is not a non negative integer.'.format(arg))
+    return int(arg)
+
+
+def checkValidSplit(arg):
+    """
+    Check if argument is a valid split for targets i.e {0...1}.
+    """
+    try:
+        if 0 <= float(arg) <= 1:
```

Comment:
Also, never use this syntax.
This will be incorrect in most languages and makes it more opaque for maintainers.
Instead, do:
```
0.0 <= float(arg) and float(arg) <= 1
```